src/main/java/org/zalando/nakadi/filters/LoggingFilter.java

```diff
@@ -134,35 +122,74 @@ protected void doFilterInternal(final HttpServletRequest request,
-
-            ACCESS_LOGGER.info("{} \"{}{}\" \"{}\" \"{}\" {} {}ms \"{}\" \"{}\" {}B",
-                    requestLogInfo.method,
-                    requestLogInfo.path,
-                    requestLogInfo.query,
-                    requestLogInfo.userAgent,
-                    requestLogInfo.user,
-                    response.getStatus(),
-                    timing,
-                    requestLogInfo.contentEncoding,
-                    requestLogInfo.acceptEncoding,
-                    requestLogInfo.contentLength);
-            nakadiKpiPublisher.publish(accessLogEventType, () -> new JSONObject()
-                    .put("method", requestLogInfo.method)
-                    .put("path", requestLogInfo.path)
-                    .put("query", requestLogInfo.query)
-                    .put("app", requestLogInfo.user)
-                    .put("app_hashed", nakadiKpiPublisher.hash(requestLogInfo.user))
-                    .put("status_code", response.getStatus())
-                    .put("response_time_ms", timing));
+    private void logRequest(final RequestLogInfo requestLogInfo, final int statusCode) {
+        final Long timing = System.currentTimeMillis() - requestLogInfo.requestTime;
+
+        logToAccessLog(requestLogInfo, statusCode, timing);
+        logToNakadi(requestLogInfo, statusCode, timing);
+        traceRequest(requestLogInfo, statusCode);
+    }
+
+    private void logToNakadi(final RequestLogInfo requestLogInfo, final int statusCode, final Long timing) {
+        nakadiKpiPublisher.publish(accessLogEventType, () -> new JSONObject()
+                .put("method", requestLogInfo.method)
+                .put("path", requestLogInfo.path)
+                .put("query", requestLogInfo.query)
+                .put("app", requestLogInfo.user)
+                .put("app_hashed", nakadiKpiPublisher.hash(requestLogInfo.user))
+                .put("status_code", statusCode)
+                .put("response_time_ms", timing));
+    }
+
+    private void logToAccessLog(final RequestLogInfo requestLogInfo, final int statusCode, final Long timing) {
+        ACCESS_LOGGER.info("{} \"{}{}\" \"{}\" \"{}\" {} {}ms \"{}\" \"{}\" {}B",
+                requestLogInfo.method,
+                requestLogInfo.path,
+                requestLogInfo.query,
+                requestLogInfo.userAgent,
+                requestLogInfo.user,
+                statusCode,
+                timing,
+                requestLogInfo.contentEncoding,
+                requestLogInfo.acceptEncoding,
+                requestLogInfo.contentLength);
+    }
+
+    private void traceRequest(final RequestLogInfo requestLogInfo, final int statusCode) {
+        if ("POST".equals(requestLogInfo.method) && requestLogInfo.path.startsWith("/event-types/") &&
+                requestLogInfo.path.contains("/events")) {
+
+            final String eventType = requestLogInfo.path.substring(13, requestLogInfo.path.lastIndexOf("/events"));
```

Comment:
Can we move the magic 13 to a constant? I guess it's the number of characters in `/event-types/`

Reply:
Yes, makes sense.
https://github.com/zalando/nakadi/pull/1027/commits/c140b6f54ec1113fdf86f8f76a4710f506dcc637